### ISSUE TITLE
XIAO NRF52840 ou XIAO NRF52840 BLE ?

### ISSUE THREAD

**Issue by jduffas** [Brezensalzer/ble_xiao_adv#1]
hello,

I tried your code with a XIAO BLE nRF52840,
When selecting "Seeed nRF52840 BLE" board, the bluefruit.h isn't found by the compiler.
but when selecting the "Seeed nRF52840" board the bluefruit.h is found, but the program can't be send to card.

is the code written for a XIAO nRF52840 BLE, or for a XIAO nRF52840 ?

thank you.

**Comment by Brezensalzer** [comment]
Hi,
Seeed has two alternative board definitions. You have to choose the "Adafruit flavor".
In the Arduino board manager it is listed as:

Seeed nRF52 boards
Version 1.0.0

The other one is based on RTOS/Arduino BLE Libs and won't work with my sketch. You should disable it and then you have only one XIAO nRF52840 variant.

**Comment by jduffas** [comment]
thank you for your answer,
but when I select the 1.1.1 version of the Seeed nRF52 boards then I tried with the 1.0.0

I got the same error:

Library Adafruit nRFCrypto has been declared precompiled:
Using precompiled library in /Users/jduffas/Library/Arduino15/packages/Seeeduino/hardware/nrf52/1.1.1/libraries/Adafruit_nRFCrypto/src/cortex-m4/fpv4-sp-d16-hard
Le croquis utilise 121124 octets (14%) de l'espace de stockage de programmes. Le maximum est de 811008 octets.
Les variables globales utilisent 14304 octets (6%) de mémoire dynamique, ce qui laisse 223264 octets pour les variables locales. Le maximum est de 237568 octets.
Upgrading target on /dev/cu.usbmodem14301 with DFU package /private/var/folders/0m/8y40g9g54199r9c5jpgw2dvr0000gn/T/arduino/sketches/2169B8E61F0A25B559250350120FE836/XIAO_BLE_VERY_LOW_CONS.ino.zip. Flow control is disabled, Single bank, Touch disabled

Failed to upgrade target. Error is: Serial port could not be opened on /dev/cu.usbmodem14301. Reason: [Errno 16] could not open port /dev/cu.usbmodem14301: [Errno 16] Resource busy: '/dev/cu.usbmodem14301'

Possible causes:
- Selected Bootloader version does not match the one on Bluefruit device.
    Please upgrade the Bootloader or select correct version in Tools->Bootloader.
- Baud rate must be 115200, Flow control must be off.
- Target is not in DFU mode. Ground DFU pin and RESET and release both to enter DFU mode.
Traceback (most recent call last):
  File "serial/serialposix.py", line 322, in open
OSError: [Errno 16] Resource busy: '/dev/cu.usbmodem14301'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "dfu/dfu_transport_serial.py", line 113, in open
  File "serial/serialutil.py", line 244, in __init__
  File "serial/serialposix.py", line 325, in open
serial.serialutil.SerialException: [Errno 16] could not open port /dev/cu.usbmodem14301: [Errno 16] Resource busy: '/dev/cu.usbmodem14301'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "__main__.py", line 296, in serial
  File "dfu/dfu.py", line 235, in dfu_send_images
  File "dfu/dfu.py", line 157, in _dfu_send_image
  File "dfu/dfu_transport_serial.py", line 115, in open
nordicsemi.exceptions.NordicSemiException: Serial port could not be opened on /dev/cu.usbmodem14301. Reason: [Errno 16] could not open port /dev/cu.usbmodem14301: [Errno 16] Resource busy: '/dev/cu.usbmodem14301'

**Comment by Brezensalzer** [comment]
Ok, so the sketch compiles error free - that's good
Arduino wants to flash the hex file via dfu and can't connect to the serial port (via USB) of the XIAO board...
* double click (really fast) on the tiny reset switch close to the USB-C connector
* a USB drive (XIAO BLE) should appear as mounted
* the serial port appears in Arduino as "/dev/cu.usbmodem21401 (Seeed XIAO nRF52840)"
* flashing via the Arduino IDE should work (be patient, needs a moment to start)

**Comment by jduffas** [comment]
Hello,
Still same problem,
I have activated the DFU on the card (I can see in and an USB)
I downgraded to 1.0.0 version of Seeed nRF52 Boards by Seeed Studio, I selected the Seeed XIAO nRF52840 board.

still the same error :


Library Adafruit nRFCrypto has been declared precompiled:
Using precompiled library in /Users/jduffas/Library/Arduino15/packages/Seeeduino/hardware/nrf52/1.0.0/libraries/Adafruit_nRFCrypto/src/cortex-m4/fpv4-sp-d16-hard
Le croquis utilise 121124 octets (14%) de l'espace de stockage de programmes. Le maximum est de 811008 octets.
Les variables globales utilisent 14304 octets (6%) de mémoire dynamique, ce qui laisse 223264 octets pour les variables locales. Le maximum est de 237568 octets.
Upgrading target on /dev/cu.usbmodem14401 with DFU package /private/var/folders/0m/8y40g9g54199r9c5jpgw2dvr0000gn/T/arduino/sketches/2169B8E61F0A25B559250350120FE836/XIAO_BLE_VERY_LOW_CONS.ino.zip. Flow control is disabled, Single bank, Touch disabled

Failed to upgrade target. Error is: Serial port could not be opened on /dev/cu.usbmodem14401. Reason: [Errno 16] could not open port /dev/cu.usbmodem14401: [Errno 16] Resource busy: '/dev/cu.usbmodem14401'

Possible causes:
- Selected Bootloader version does not match the one on Bluefruit device.
    Please upgrade the Bootloader or select correct version in Tools->Bootloader.
- Baud rate must be 115200, Flow control must be off.
- Target is not in DFU mode. Ground DFU pin and RESET and release both to enter DFU mode.
Traceback (most recent call last):
  File "serial/serialposix.py", line 322, in open
OSError: [Errno 16] Resource busy: '/dev/cu.usbmodem14401'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "dfu/dfu_transport_serial.py", line 113, in open
  File "serial/serialutil.py", line 244, in __init__
  File "serial/serialposix.py", line 325, in open
serial.serialutil.SerialException: [Errno 16] could not open port /dev/cu.usbmodem14401: [Errno 16] Resource busy: '/dev/cu.usbmodem14401'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "__main__.py", line 296, in serial
  File "dfu/dfu.py", line 235, in dfu_send_images
  File "dfu/dfu.py", line 157, in _dfu_send_image
  File "dfu/dfu_transport_serial.py", line 115, in open
nordicsemi.exceptions.NordicSemiException: Serial port could not be opened on /dev/cu.usbmodem14401. Reason: [Errno 16] could not open port /dev/cu.usbmodem14401: [Errno 16] Resource busy: '/dev/cu.usbmodem14401'
